src/api/routes/offer.js

```diff
@@ -58,6 +58,27 @@ module.exports = (app) => {
         }
     });
 
+    /**
+     * Gets all the offers of a certain company from the db
+     */
+    router.get("/company/:companyID", async (req, res, next) => {
+        try {
+            const offers = await (new OfferService()).getOfferByCompanyID(req.params.companyID, req.user, req.hasAdminPrivileges);
+
+            if (!offers) {
+                return next(new APIError(
+                    HTTPStatus.NOT_FOUND,
+                    ErrorTypes.FORBIDDEN,
+                    ValidationReasons.COMPANY_OFFER_NOT_FOUND(req.params.companyID) // This is not the offerId
+                ));
```

Comment:
This will only happen if `offers` is undefined, meaning that something went wrong in the find call. 
If this error message is necessary, then you also need to verify if the array is empty

Reply:
Thinking back on it, I believe this verification is not needed anyway so I removed it. I had it since I used another route as a template.